### ISSUE TITLE
Print report path

### ISSUE THREAD

**Issue by hasnainjaved** [JLLeitschuh/ktlint-gradle#51]
Hello,

Is there a way to print out the path for the report file on the console ? Useful for CI/CD environments.
If not, is there a work around to print it out ?

Kind Regards .

**Comment by JLLeitschuh** [comment]
Umm... Currently no. I don't think we expose that. We could/should. This would probably be easiest to do if we were to make the Ktlint task extend something like [`SourceTask`](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/SourceTask.html) and then add our own custom fields to it like `reportDirectory`. That way API consumers could just read from that field.
@Tapchicoma thoughts?

**Comment by Tapchicoma** [comment]
@hasnainjaved can you provide examples of use-case?

Currently plugin always create reports in `$project.buildDir/reports/ktlint`. Can your CI/CD just grab all the files from that directory?

**Comment by Tapchicoma** [comment]
I don't see the point of adding this feature, but possible solution may be:
check for `verbose` flag value in extension and on `true` print `"Ktlint report file: ${fileLocation}"` to `project.logger.warn`

**Comment by hasnainjaved** [comment]
@Tapchicoma since CI/CD servers don't have a GUI so grabbing the path from the console and pasting into a browser would be convenient eg. Maven's surefire plugin prints the path of the report in case of test failures. We are currently doing what you have suggested.

@JLLeitschuh Just printing the path would suffice.

**Comment by JLLeitschuh** [comment]
I don't think we should just print the directory. We should make it something that the user can retrieve as a method or field. Then they can do what they want with it.

**Comment by Tapchicoma** [comment]
@hasnainjaved just to clarify:

You are using some non-Gradle tool to post failed results back to PR as a comment? Not failed builds should not print report path?

Will some Gradle callback suit your goals?

**Comment by Tapchicoma** [comment]
@hasnainjaved is this issue still actual?

**Comment by eskatos** [comment]
I'd say that in case of a linting failure it would be nice to have the report(s) path(s) in the exception message so that Gradle will display them in the build failure summary. This will make it work nicely even with `--parallel` and `--continue` and multiple tasks failure.

Then, logging the report(s) path(s) at e.g. `info` level should be enough for folks that would like to get the path always, they would simply have to use `-i`.

**Comment by Tapchicoma** [comment]
Related commit in KtLint: https://github.com/shyiko/ktlint/commit/30ec67d88c3188609f506df1dd4b59ff2ada630f

**Comment by Tapchicoma** [comment]
New released version of KtLint `0.24.0` prints report location in console:
```
"plain" report written to /home/user/app/build/reports/ktlint/ktlint-release.txt
```
